### PR TITLE
python snippets

### PR DESCRIPTION
**Pull request by ubbabeck**

### CODE REVIEW

**Review by erdemyerebasmaz — APPROVED** [review]
LGTM

**Comment by ubbabeck** [comment]
Since the latest pr, I added a CI which checks the format of the code and also testes it. If we wish to test it we need to add the following secrets `API_KEY` and `MNEMONIC`. 

The CI fails now since the secrets aren't added yet, but succeeds in my forked repo.

File: snippets/python_snippets/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+version = "0.2.7"
+description = ""
+authors = ["ruben beck <ruben.beck@pm.me>"]
+license = "MIT"
+readme = "README.md"
+
+[tool.poetry.dependencies]
+python = "^3.8"
+ruff = "^0.1.4"
+breez-sdk = { git = "https://github.com/cnixbtc/breez-sdk.git", branch = "pypi", subdirectory = "libs/sdk-bindings/bindings-python" }
```
Comment (ok300) [inline snippets/python_snippets/pyproject.toml:12]: Do we need to use this fork? Could it not use our bindings from the main repo?

**Review by ok300 — COMMENTED** [review]
Looks promising!

Technically, the CI tests don't have to run any SDK call, they just have to make sure the py files have valid python code.

Could the CI step be changed to only check that? Somehow import or load all snippet files and return success?

File: snippets/python_snippets/pyproject.toml (hunk 1 above)
Comment (ubbabeck) [reply snippets/python_snippets/pyproject.toml]: I removed it since we use the temporary package from test pypi.

**Comment by ubbabeck** [comment]
> Looks promising!
> 
> Technically, the CI tests don't have to run any SDK call, they just have to make sure the py files have valid python code.
> 
> Could the CI step be changed to only check that? Somehow import or load all snippet files and return success?

I changed it to load and compile all the code, but I does not assure us that we have not made any type errors, which I think only can be caught at runtime.

**Comment by erdemyerebasmaz** [comment]
@ubbabeck ok300-snippets-rust has been merged and there has been additional changes on top of it on master. I've changed the base branch of this PR from ok300-snippets-rust to main. Please address any conflicts that may come up.

**Comment by ok300** [comment]
> I changed it to load and compile all the code, but I does not assure us that we have not made any type errors, which I think only can be caught at runtime.

I see.

Maybe that is good enough for CI checks, and to test the python snippets for correct types we'd have to run locally a script like:

```bash
#!/bin/bash

export API_KEY="..."
export MNEMONIC="..."

cd snippets/python_snippets
python main.py $API_KEY $MNEMONIC
```

What do you think?

**Comment by ubbabeck** [comment]
> I see.
> 
> Maybe that is good enough for CI checks, and to test the python snippets for correct types we'd have to run locally a script like:
> 
> ```shell
> #!/bin/bash
> 
> export API_KEY="..."
> export MNEMONIC="..."
> 
> cd snippets/python_snippets
> python main.py $API_KEY $MNEMONIC
> ```
> 
> What do you think?

 @ok300 Sounds good to me 👍

**Comment by JssDWt** [comment]
If you rebase onto main, the CI should work again

**Pull request merged**